test/distributed/_shard/sharding_plan/test_sharding_plan.py

```diff
@@ -97,6 +91,10 @@ def test_basic_sharding_plan(self):
             # make sure we can run sharded computation
             input = torch.rand(22, 17).cuda(self.rank)
             sharded_output = megatron_lm(input)
+            local_output = local_megatron_lm(input)
+
+            # verify and make sure local and sharded output matches
+            self.assertEqual(local_output, sharded_output)
```

Comment:
Probably better to have a full forward + backward + optimizer pass for local and sharded and then verify the result.

Can probably enhance this in a separate PR, but having a end to end training loop with sharder is just good validation in general.